src/main/java/user_code_resolver/UserCodeResolver.java

```diff
@@ -0,0 +1,263 @@
+package user_code_resolver;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.javers.core.Javers;
+import org.javers.core.JaversBuilder;
+import org.javers.core.diff.Change;
+import org.javers.core.diff.Diff;
+import org.javers.core.diff.ListCompareAlgorithm;
+import org.javers.core.diff.changetype.container.ContainerElementChange;
+import org.javers.core.diff.changetype.container.ElementValueChange;
+import org.javers.core.diff.changetype.container.ListChange;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * class to save user code between generations
+ *
+ * @author Jonas Knebel
+ * @version 2.0 improved Structure of file and fixed some errors
+ */
+public class UserCodeResolver {
+
+
+    private final File file = new File("userCode.json");
+    private static final Logger logger = LogManager.getLogger(UserCodeResolver.class);
+
+    public File getFile() {
+        return file;
+    }
+
+    /**
+     * Instantiates a new User code resolver.
+     */
+    public UserCodeResolver() {
+    }
+
+    /**
+     * Write the JSOn Object in the new Project
+     *
+     * @param project
+     * @param folder
+     * @throws IOException
+     */
+    public void writeUserContentInNewProject(Project project, File folder) throws IOException {
+        if (project.getFiles().size() > 0) {
+            updateNamesOfImplFiles(project, folder);
+        }
+
+        for (UserFile file : project.getFiles()) {
+            Path path = Paths.get(file.getFilename());
+            if (Files.notExists(path)) {
+                Files.createFile(path);
+            }
+            if (file.getFilename().endsWith(".jpg") || file.getFilename().endsWith(".jpeg") || file.getFilename().endsWith(".png") || file.getFilename().endsWith(".gif")) {
```

Comment:
Kannst du hier nicht die selbe regular expression wie in Zeile 114 nehmen?

Reply:
jo good catch